doc/release-notes.rst

```diff
@@ -30,6 +30,12 @@ Documentation
 * Documentation refactorization with examples. :issue:`472` by :user:`Guillaume
   Lemaitre <glemaitre>`.
 
+Maintenance
+...........
+
+* Remove :class:`pydicom.filereader.DicomIter` since it is unused. :issue:`482`
+  by :user:`Guillaume Lemaitre <glemaitre>`.
+
```

Comment:
Uhm in 0.9.8 you have some (I did not check for the latest one). What we should absolutely is to agree on something which we should follow from 1.0.0. I would be incline to state only the PR and to add this inside the contributing guide.

I also think that we should split the what's new in several notes. This is easier to maintain